### ISSUE TITLE
By default, return a URL to each artefact where possible

### ISSUE THREAD

**Issue by MeltedHyperion7** [robota-suite/robota-core#2]
[Copied from Gitlab.]

It is common to have to create reports using the information extracted from the projects we work with.  It's generally better for usability of the reports if when we mention an artefact we also provide a link to go and inspect it.  (This certainly happens a lot in RoboTA.)  At the moment, we have to do a lot of faffing about to create these links, from a mixture of information provided by RoboTA Core and other information.  This would be much simpler if RoboTA Core was in the habit of returning a URL along with the artefact itself.

**Comment by suzanneEmbury** [comment]
Yes - we added in the URL based on where it was needed in the early versions of the code, but that isn't super helpful.  It would be good if we just had a model that included the URL of each project object where it can easily be acquired or constructed.  That would make things easier for people building user interfaces/reports with RoboTA Core.

It would be great to see your ideas on adding this, @MeltedHyperion7 .

**Comment by MeltedHyperion7** [comment]
@suzanneEmbury I've raised a PR to add URLs to `Branch` and `Diff`.

Things to note:
- Local repositories have URLs set to `None`.
- Gitlab diff URLs point to the specific file in the diff.
- GitHub diff URLs point to the diff page but not to the specific file. I don't think there is a way to achieve this with the current GitHub UI.

**Comment by suzanneEmbury** [comment]
Thanks @MeltedHyperion7 - this is great.  I will definitely look at the PR you've created but am a bit swamped with marking and other work at the moment, so I may not be able to respond very quickly.  But I'll definitely get to it!
